src/py42/clients/securitydata.py

```diff
@@ -397,14 +402,32 @@ def _get_plans_in_node(destination, node):
     }
 
 
-def _parse_file_location_response(response):
-
-    for location in response[u"locations"]:
-        paths = []
+def _parse_file_location_response(locations):
+    devices = {}
+    for location in locations:
         file_name = location[u"fileName"]
+        file_path = u"{}{}".format(location[u"filePath"], file_name)
         device_id = location[u"deviceUid"]
-        paths.append(u"{}{}".format(location[u"filePath"], file_name))
-        yield device_id, paths
+        device_entry = devices.get(device_id)
+        if device_entry:
+            device_entry["paths"].append(file_path)
+        else:
+            devices[device_id] = {"deviceGuid": device_id, "paths": [file_path]}
+
+    return [devices[key] for key in devices]
```

Comment:
`return devices.values()`  or `devices.append({"deviceGuid".....}) return devices` where `devices = []`

`device_entry` is being defined but not being used, I guess 
 `devices[device_id] = {"deviceGuid": device_id, "paths": [file_path]}` should be changed to  `devices[device_id] = {"deviceGuid": device_id, "paths": device_entry["paths"]}`

Reply:
This implementation is correct, but I should change line 413 to:
`devices.get(device_id)["paths"]` to be more clear. That line is modifying the `devices` dictionary, even though it kind of doesn't look like it is.

Reply:
I second changing line 413